.github/workflows/pr.yml

```diff
@@ -10,8 +10,15 @@ jobs:
   mypy:
     needs: ruff
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11", "3.12", "3.13"]
     steps:
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
       - uses: actions/checkout@v3
+      - run: sudo apt install libsystemd-dev
       - run: pip install -r requirements.txt
-      - run: pip install mypy
-      - run: mypy --install-types --non-interactive --explicit-package-bases --disable-error-code import-untyped .
```

Comment:
`--disable-error-code import-untyped` isn't valid for earlier versions of mypy, which are in turn necessary to support python 3.7